backend/assignments/1/src/main/java/org/example/CoinList.java

```diff
@@ -0,0 +1,133 @@
+package org.example;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class CoinList {
+    private TraderList traderList;
+    public CoinList(TraderList traderList) {
+        this.traderList = traderList;
+    }
+    public static List<Coin> coinArray = new CopyOnWriteArrayList<>();
+    private static final LoggerFile log = new LoggerFile();
+    public static void readCoinCSV(String filePath) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath))) {
+            String line = reader.readLine();
+            while ((line = reader.readLine()) != null) {
+                String[] data = line.split(",");
+                Coin coin = createCoinFromData(data);
+                coinArray.add(coin);
+            }
+        }
+        catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static Coin createCoinFromData(String[] data) {
+        Coin coin = new Coin();
+        try {
+            coin.setRank(Integer.parseInt(data[1]));
+        } catch (NumberFormatException e) {
+            coin.setRank(0);
+        }
+
+        coin.setName(data[2]);
+        coin.setSymbol(data[3]);
+
+        try {
+            coin.setPrice(Double.parseDouble(data[4]));
+        } catch (NumberFormatException e) {
+            coin.setPrice(0.0);
+        }
+
+        try {
+            coin.setSupply(Long.parseLong(data[5]));
+        } catch (NumberFormatException e) {
+            coin.setSupply(0);
+        }
+
+        return coin;
+    }
+
+    public void retrieveCoinDetailsByName(String s) {
```

Comment:
Lets not use generic variable names like s, c etc. Try to make it more readable.